File: online-bookstore-server/microservice-bookstore-common/src/main/java/com/onlinebookstore/entity/bookserver/BookType.java
package com.onlinebookstore.entity.bookserver;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.io.Serializable;

/**
 * @author rkc
 * @date 2020/9/30 16:08
 * @version 1.0
 */
@Data
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class BookType implements Serializable {

    /**
     * 自增长id
     */
    private Integer id;

    /**
     * 类型名称
     */
    private String type;

    /**
     * 说明
     */
    private String supplement;

    /**
     * 图标
     */
    private String img;
}

File: online-bookstore-android/app/src/main/java/com/rkc/onlinebookstore/model/book/BookType.kt
package com.rkc.onlinebookstore.model.book

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

/**
 * @author rkc
 * @date 2020/11/14 16:39
 * @version 1.0
 */
@Parcelize
data class BookType(
    @SerializedName("id") val id: Int = -1,
    @SerializedName("type") val type: String = "",
    @SerializedName("supplement") val supplement: String = "",
    @SerializedName("img") val imgIcon: String = "",
    @SerializedName("books") val books: ArrayList<Book> = arrayListOf()
) : Parcelable